yt/yt/server/scheduler/node_shard.cpp

```diff
@@ -597,17 +597,10 @@ void TNodeShard::DoProcessHeartbeat(const TScheduler::TCtxNodeHeartbeatPtr& cont
     }
 
     bool skipScheduleAllocations = false;
-    if (hasWaitingAllocations || isThrottlingActive) {
```

Comment:
Let's remove hasWaitingAllocations completely

Reply:
We need it to pass it by reference to `ProcessHeartbeatAllocations` on line 596

Reply:
It is true, but look, what we pass it in ProcessHeartbeatAllocations for :)

Reply:
Removed it from ProcessHeartbeatAllocations too